vizro-core/tests/unit/vizro/models/_action/test_action.py

```diff
@@ -0,0 +1,288 @@
+        assert action.outputs == []
+
+    def test_create_action_mandatory_and_optional(self, test_action_function):
+        inputs = ["component_1.property_A", "component_1.property_B"]
+        outputs = ["component_2.property_A", "component_2.property_B"]
+
+        action = Action(function=test_action_function, inputs=inputs, outputs=outputs)
+
+        assert hasattr(action, "id")
+        assert action.function == test_action_function
+        assert action.inputs == inputs
+        assert action.outputs == outputs
+
+    @pytest.mark.parametrize(
+        "inputs, outputs",
+        [
+            ([], []),
+            (["component.property"], ["component.property"]),
+            (["component.property", "component.property"], ["component.property", "component.property"]),
+            (
+                ["Component_1.Property_A", "Component_2.Property_B"],
+                ["Component_1.Property_A", "Component_2.Property_B"],
+            ),
+        ],
+    )
+    def test_inputs_outputs_valid(self, inputs, outputs, test_action_function):
+        action = Action(function=test_action_function, inputs=inputs, outputs=outputs)
+
+        assert action.inputs == inputs
+        assert action.outputs == outputs
+
+    @pytest.mark.parametrize(
+        "inputs",
+        [
+            "string",
+            "component_property",
+            "compo-nent.property",
+            "component.property_1",
+        ],
+    )
+    def test_inputs_invalid(self, inputs, test_action_function):
+        with pytest.raises(ValidationError, match="value is not a valid list"):
+            Action(function=test_action_function, inputs=inputs, outputs=[])
+
+    @pytest.mark.parametrize(
+        "outputs",
+        [
+            "string",
+            "component_property",
+            "compo-nent.property",
+            "component.property_1",
+        ],
+    )
+    def test_outputs_invalid(self, outputs, test_action_function):
+        with pytest.raises(ValidationError, match="value is not a valid list"):
+            Action(function=test_action_function, inputs=[], outputs=outputs)
+
+    @pytest.mark.parametrize("file_format", [None, "csv", "xlsx"])
+    def test_export_data_file_format_valid(self, file_format):
+        action = Action(id="action_test", function=export_data(file_format=file_format))
+        assert action.id == "action_test"
+        assert action.inputs == []
+        assert action.outputs == []
+
+    def test_export_data_file_format_invalid(self):
+        with pytest.raises(
+            ValueError, match='Unknown "file_format": invalid_file_format.' ' Known file formats: "csv", "xlsx".'
+        ):
+            Action(function=export_data(file_format="invalid_file_format"))
+
+    def test_export_data_xlsx_without_required_libs_installed(self, monkeypatch):
+        monkeypatch.setitem(sys.modules, "openpyxl", None)
+        monkeypatch.setitem(sys.modules, "xlswriter", None)
+        # monkeypatch.setattr(
```

Comment:
Do we not want to eremove the commented out code @petar-qb ?

Reply:
Sure, my mistake 😞.
Let's patch it in upcoming PR.

Reply:
Also `# return action_components` in _action.py 🙂